### PR TITLE
[Fix] proper validation and handling of Latest schema version in Kafk…

### PR DESCRIPTION
**Pull request by mproch**
This fixes a couple of things:
- handling validation of "Last schema version"
- validation of output in sink
- handling some of the edge cases that may happen
- small refactoring

### CODE REVIEW

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala [hunk 1]
```diff
@@ -21,15 +29,29 @@ trait KafkaAvroBaseTransformer[T, Y] extends SingleInputGenericNodeTransformatio
 
   override type State = Nothing
 
-  protected def topicParam: Parameter = Parameter[String](KafkaAvroFactory.TopicParamName).copy(editor = Some(FixedValuesParameterEditor(
-    schemaRegistryClient.getAllTopics.getOrElse(Nil).sorted.map(v => FixedExpressionValue(s"'$v'", v))
-  )))
+  protected def topicParam(implicit nodeId: NodeId): WithError[Parameter] = {
+    val topics = schemaRegistryClient.getAllTopics
```
Comment (lciolecki) [inline engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala:33]: I looked to CachedConfluentSchemaRegistryClient and I think we should:

* Change topicsCache to cache, this should be some grouping cache
* At getAllTopics move fixedKey to const and set value `all-topics`

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/source/KafkaAvroSourceFactory.scala [hunk 2]
```diff
@@ -21,20 +22,29 @@ class KafkaAvroSourceFactory[T:TypeInformation](val schemaRegistryProvider: Sche
 
   override def contextTransformation(context: ValidationContext, dependencies: List[NodeDependencyValue])
                                     (implicit nodeId: ProcessCompilationError.NodeId): NodeTransformationDefinition = {
-    case TransformationStep(Nil, _) => NextParameters(initialParameters)
-    case TransformationStep((KafkaAvroFactory.TopicParamName, DefinedEagerParameter(topic:String, _)) :: Nil, _) =>
-       NextParameters(List(versionParam(topic)), Nil, None)
-    case TransformationStep((KafkaAvroFactory.TopicParamName, DefinedEagerParameter(topic:String, _)) ::
-      (KafkaAvroFactory.SchemaVersionParamName, DefinedEagerParameter(version:Integer, _)) ::Nil, _) =>
-      val result = createSchemaRegistryProvider(topic, version).typeDefinition
+    case TransformationStep(Nil, _) =>
```
Comment (lciolecki) [inline engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/source/KafkaAvroSourceFactory.scala:25]: Does this code is duplicated at `KafkaAvroSinkFactory`? Maybe we can extract it to some Mixin..?

File: engine/flink/avro-util/src/test/scala/pl/touk/nussknacker/engine/avro/sink/KafkaAvroSinkFactoryValidationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import pl.touk.nussknacker.engine.compile.nodevalidation.{GenericNodeTransformationValidator, TransformationResult}
+import pl.touk.nussknacker.engine.expression.ExpressionEvaluator
+import pl.touk.nussknacker.engine.graph.evaluatedparam.Parameter
+import pl.touk.nussknacker.engine.graph.expression.Expression
+import pl.touk.nussknacker.engine.spel.Implicits._
+import pl.touk.nussknacker.engine.testing.{EmptyProcessConfigCreator, LocalModelData}
+
+import scala.collection.JavaConverters._
+
+class KafkaAvroSinkFactoryValidationSpec extends FunSuite with Matchers with KafkaAvroSinkSpecMixin {
```
Comment (lciolecki) [inline engine/flink/avro-util/src/test/scala/pl/touk/nussknacker/engine/avro/sink/KafkaAvroSinkFactoryValidationSpec.scala:24]: Why not in `KafkaAvroSinkFactorySpec.scala`?

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala (hunk 1 above)
Comment (lciolecki) [reply engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala:33]: I know it's not part of your changes, but maybe you can refactor alsow this? :)

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/source/KafkaAvroSourceFactory.scala (hunk 2 above)
Comment (mproch) [reply engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/source/KafkaAvroSourceFactory.scala:25]: I extracted a bit

File: engine/flink/avro-util/src/test/scala/pl/touk/nussknacker/engine/avro/sink/KafkaAvroSinkFactoryValidationSpec.scala (hunk 3 above)
Comment (mproch) [reply engine/flink/avro-util/src/test/scala/pl/touk/nussknacker/engine/avro/sink/KafkaAvroSinkFactoryValidationSpec.scala]: True, merged

File: engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala (hunk 1 above)
Comment (mproch) [reply engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala:33]: refactored a bit :)

**Pull request merged**